src/Http/Response.php

```diff
@@ -925,8 +925,11 @@ public function getStatusCode()
      */
     public function withStatus($code, $reasonPhrase = '')
     {
+        if (!isset($this->_statusCodes[$code])) {
+            throw new InvalidArgumentException('Unknown status code');
+        }
         $new = clone $this;
-        $new->_status = $code;
+        $new->_status = $code === null ? $this->_status : $code;
```

Comment:
Due to the check and exception added above, the `$code === null` case will never occur, hence this check is not needed.

Reply:
@ADmad Haven't had my coffee yet... Removed the check, the check earlier in the method should suffice.